src/vos/vos_obj_index.c

```diff
@@ -623,7 +623,11 @@ oi_iter_fetch(struct vos_iterator *iter, vos_iter_entry_t *it_entry,
 	}
 	it_entry->ie_child_type = VOS_ITER_DKEY;
 
-	it_entry->ie_last_update = obj->vo_max_write;
+	/* Upgrading case, set it to current epoch */
+	if (obj->vo_max_write == 0)
+		it_entry->ie_last_update = epr.epr_hi;
```

Comment:
This is incorrect, you'd call vos_ilog_last_update() to get the last update epoch.